docgen/src/guide/Conditional_display.md

```diff
@@ -6,89 +6,65 @@ category: guide
 navWeight: 40
 ---
 
-Using our connector and [`createConnector`](guide/Custom_connectors.html) approach, you can conditionally display content based on the search state.
+When no results are found you might want to display some specific contents helping the user go back
+to a search that was successful. 
+
+To help you do conditional rendering based on the `searchState` and the 
+`searchResults` of InstantSearch, we provide the [`connectStateResults`](connectors/connectStateResults.html) connector. 
 
 ## Displaying content when the query is empty
 
 ```jsx
-const Content = createConnector({
-    displayName: 'ConditionalQuery',
-    getProvidedProps(props, searchState) {
-      return {query: searchState.query};
-    },
- })(({query}) => {
-    const content = query
-      ? <div>The query {query} exists</div>
-      : <div>No query</div>;
-    return <div>{content}</div>;
- });
+const Content = connectStateResults(
+  ({ searchState }) =>
```

Comment:
I'm wondering if we can't call this: `({state, results})`

Reply:
It's called `searchState` and `searchResults` everywhere especially in the documentation when looking for the structure so I'd rather keep the same name. If a user wants to call them state or results, he can definitely do it.